src/StreamJsonRpc/MessageFormatterHelper.cs

```diff
@@ -0,0 +1,24 @@
+﻿
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    public class MessageFormatterHelper
+    {
+        public static Type FindIProgressOfT(Type objectType)
+        {
+            if (objectType.IsConstructedGenericType && objectType.GetGenericTypeDefinition().Equals(typeof(IProgress<>)))
+            {
+                return objectType;
+            }
+            else if (objectType.GetTypeInfo().GetInterfaces().Any(i => i.IsConstructedGenericType && i.GetGenericTypeDefinition() == typeof(IProgress<>)))
+            {
+                return typeof(IProgress<>).MakeGenericType(objectType.GenericTypeArguments);
```

Comment:
>typeof(IProgress<>).MakeGenericType(objectType.GenericTypeArguments [](start = 23, length = 67)

This returns a franken-type. if `objectType` is `MyClass<Seed>` which implements `IProgress<Tree>`, then this will create an `IProgress<Seed>`. And if `MyClass` isn't even a generic type, it will throw.
If the point is to return the `IProgress<T>` that you found, you can do so by changing `Any` to `FirstOrDefault` and return its result directly.